boutdata/tests/test_collect.py

```diff
@@ -0,0 +1,2773 @@
+            if x in collect_kwargs:
+                collect_kwargs.pop(x)
+        # Never remove x-boundaries when collecting from a squashed file without them
+        collect_kwargs["xguards"] = True
+        # Never remove y-boundaries when collecting from a squashed file without them
+        collect_kwargs["yguards"] = "include_upper"
+
+    for varname in expected:
+        actual = collect(varname, path=path, **collect_kwargs)
+        npt.assert_array_equal(expected[varname], actual)
+        actual_keys = list(actual.attributes.keys())
+        if varname in expected_attributes:
+            for a in expected_attributes[varname]:
+                assert actual.attributes[a] == expected_attributes[varname][a]
+                actual_keys.remove(a)
+
+        if "fieldperp" in varname:
+            assert actual.attributes["yindex_global"] == fieldperp_global_yind
+            actual_keys.remove("yindex_global")
+
+        assert actual_keys == ["bout_type"]
+
+        if "field3d_t" in varname:
+            assert actual.attributes["bout_type"] == "Field3D_t"
+        elif "field3d" in varname:
+            assert actual.attributes["bout_type"] == "Field3D"
+        elif "field2d_t" in varname:
+            assert actual.attributes["bout_type"] == "Field2D_t"
+        elif "field2d" in varname:
+            assert actual.attributes["bout_type"] == "Field2D"
+        elif "fieldperp_t" in varname:
+            assert actual.attributes["bout_type"] == "FieldPerp_t"
+        elif "fieldperp" in varname:
+            assert actual.attributes["bout_type"] == "FieldPerp"
+        elif "_t" in varname or varname == "t_array":
+            assert actual.attributes["bout_type"] == "scalar_t"
+        else:
+            assert actual.attributes["bout_type"] == "scalar"
+
+
+class TestCollect:
+    @pytest.mark.parametrize("squash", [False, True])
+    @pytest.mark.parametrize("collect_kwargs", collect_kwargs_list)
+    def test_core_min_files(self, tmp_path, squash, collect_kwargs):
+        grid_info = {}
+        grid_info["iteration"] = 6
+        grid_info["MXSUB"] = 3
+        grid_info["MYSUB"] = 4
+        grid_info["MZSUB"] = 5
+        grid_info["MXG"] = 2
+        grid_info["MYG"] = 2
+        grid_info["MZG"] = 0
+        grid_info["NXPE"] = 1
+        grid_info["NYPE"] = 1
+        grid_info["NZPE"] = 1
+        grid_info["nx"] = grid_info["NXPE"] * grid_info["MXSUB"] + 2 * grid_info["MXG"]
+        grid_info["ny"] = grid_info["NYPE"] * grid_info["MYSUB"]
+        grid_info["nz"] = grid_info["NZPE"] * grid_info["MZSUB"]
+        grid_info["MZ"] = grid_info["nz"]
+        grid_info["ixseps1"] = 7
+        grid_info["ixseps2"] = 7
+        grid_info["jyseps1_1"] = -1
+        grid_info["jyseps2_1"] = grid_info["ny"] // 2 - 1
+        grid_info["ny_inner"] = grid_info["ny"] // 2
+        grid_info["jyseps1_2"] = grid_info["ny"] // 2 - 1
+        grid_info["jyseps2_2"] = grid_info["ny"] // 2
```

Comment:
It looks like `grid_info` is only slightly modified from time to time. Would it make sense to make `grid_info` a class member and let each test make the needed modifications on a copy?

Reply:
Good point. There was a small amount of logic (depending on number of X-points) in creating `grid_info`, so I've added a utility method to construct the grid_info dicts. Reduces the number of lines quite a bit :+1: